branchLayoutImpl/src/main/java/com/virtuslab/branchlayout/impl/readwrite/BranchLayoutFileUtils.java

```diff
@@ -32,4 +35,17 @@ public static IndentSpec getDefaultSpec() {
         e -> new BranchLayoutException("Error while loading branch layout file (${path.toAbsolutePath()})", e));
   }
 
+  public static boolean hasProperIndentationCharacter(String line, char expectedIndentationCharacter) {
+    char unexpectedIndentationCharacter = expectedIndentationCharacter == SPACE_INDENT_CHARACTER
+        ? TAB_INDENT_CHARACTER
+        : SPACE_INDENT_CHARACTER;
+    for (int i = 0; i < line.length(); i++) {
```

Comment:
You can `for (var currentCharacter : line.toCharArray()) {` here.
Or even better...
`return Stream.ofAll(branchName.toCharArray()).takeWhile(c -> c != expectedIndentationCharacter).headOption().map(c -> c != unexpectedIndentationCharacter).getOrElse(true);`

Reply:
Nice! Very nice! :heart: